util/topsql/reporter/datamodel.go

```diff
@@ -693,18 +708,26 @@ func (m *normalizedPlanMap) take() *normalizedPlanMap {
 }
 
 // toProto converts the normalizedPlanMap to the corresponding protobuf representation.
-func (m *normalizedPlanMap) toProto(decodePlan planBinaryDecodeFunc) []tipb.PlanMeta {
+func (m *normalizedPlanMap) toProto(decodePlan planBinaryDecodeFunc, compressPlan planBinaryCompressFunc) []tipb.PlanMeta {
 	metas := make([]tipb.PlanMeta, 0, m.length.Load())
 	m.data.Load().(*sync.Map).Range(func(k, v interface{}) bool {
-		planDecoded, errDecode := decodePlan(v.(string))
-		if errDecode != nil {
-			logutil.BgLogger().Warn("[top-sql] decode plan failed", zap.Error(errDecode))
+		originalMeta := v.(planMeta)
+		protoMeta := tipb.PlanMeta{
+			PlanDigest: hack.Slice(k.(string)),
+		}
+
+		var err error
+		if originalMeta.isLarge {
+			protoMeta.EncodedNormalizedPlan = compressPlan(hack.Slice(originalMeta.binaryNormalizedPlan))
+		} else {
+			protoMeta.NormalizedPlan, err = decodePlan(originalMeta.binaryNormalizedPlan)
+		}
```

Comment:
Why do we use `isLarge` instead of putting `if len(binaryNormalizedPlan) > MaxBinaryPlanSize` directly here?

Reply:
Can dependency issue convince you?

Reply:
Of course, this can't be considered a "problem", I'm just interested why this is

Reply:
The decision-maker and the performer are separated. The performer just carries out the order without taking responsibility.